### DISCUSSION TITLE
Clear IIO Buffer

### DISCUSSION THREAD

**Discussion by tylersweat** [analogdevicesinc/libiio#1161] [Q&A]
Hi, 

I'm using libiio with the PyADI-IIO Python wrapper for the PlutoSDR. Is there a way to clear the receive buffer so that I know that when I get data from the buffer that it is fresh?

Alternatively, is there a way to set it up so that new data replaces the old data as it becomes available?

**Comment by tfcollins** [comment]
@tylersweat the easiest way to do this is to set the kernel buffers to 1, then pull 2 buffers each time to guarantee "fresh" data. Throw away the first. Due to the way the buffering works between the kernel and userspace this I believe is the only way to accomplish what you want.
